data/src/main/java/org/apache/iceberg/data/GeneratePartitionStats.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.data;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.apache.iceberg.ImmutableGenericPartitionStatisticsFile;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.PartitionStatisticsFile;
+import org.apache.iceberg.PartitionStatsUtil;
+import org.apache.iceberg.Partitioning;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.types.Comparators;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.SnapshotUtil;
+import org.apache.iceberg.util.Tasks;
+import org.apache.iceberg.util.ThreadPools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GeneratePartitionStats {
+  private static final Logger LOG = LoggerFactory.getLogger(GeneratePartitionStats.class);
+
+  private final Table table;
+  private String branch;
+
+  public GeneratePartitionStats(Table table) {
+    this.table = table;
+  }
+
+  public GeneratePartitionStats(Table table, String branch) {
+    this.table = table;
+    this.branch = branch;
+  }
+
+  /**
+   * Computes the partition stats for the current snapshot and writes it into the metadata folder.
+   *
+   * @return {@link PartitionStatisticsFile} for the latest snapshot id or null if table doesn't
+   *     have any snapshot.
+   */
+  public PartitionStatisticsFile generate() {
+    Snapshot currentSnapshot = SnapshotUtil.latestSnapshot(table, branch);
+    if (currentSnapshot == null) {
+      Preconditions.checkArgument(
+          branch == null, "Couldn't find the snapshot for the branch %s", branch);
```

Comment:
No. Empty branches will have a dummy snapshot without manifest files. 

https://github.com/apache/iceberg/blob/76dba8fe83c4496318cc34436e610cf50f43054d/core/src/main/java/org/apache/iceberg/SnapshotManager.java#L81

So, I am not throwing exception for that case. 
Added test and It creates an empty stats file for the dummy snapshot since snapshot exist.